src/utils/utils.py

```diff
@@ -187,108 +190,75 @@ def load_lines(filename: str) -> Iterable[str]:
             yield line.strip()
 
 
-def load_diagnostic_tsv(
+def load_diagnostic_tsv_and_index(
         data_file,
         max_seq_len,
+        label_idx,
         s1_idx=0,
         s2_idx=1,
-        targ_idx=2,
-        idx_idx=None,
-        targ_map=None,
-        targ_fn=None,
+        label_fn=None,
         skip_rows=0,
-        delimiter='\t',
-        filter_idx=None,
-        filter_value=None):
-    '''Load a tsv
-
-    It loads the data with all it's attributes from diagnostic dataset for MNLI'''
+        delimiter='\t'):
+    '''Load a tsv and  indexes the columns from the diagnostic tsv.
+        This is only used for MNLI-diagnostic right now.
+
+    Args:
+        data_file: string
+        max_seq_len: int
+        s1_idx: int
+        s2_idx: int
+        label_fn: function
+        skip_rows: list of ints
+        delimiter: string
+
+    Returns:
+        A dictionary of the necessary indexed fields, the tokenized sent1 and sent2
```

Comment:
It seems like a leaky abstraction to do the indexing in this function - can we do the indexing layer as with other tasks, and use the standard indexing/vocab code?

Reply:
I guess this functionality was here before, so fine to leave a TODO to clean it up.

Reply:
Yep, I'll leave as TODO

Reply:
I don't really know why this is a separate function...I think Jan Hula authored this?